metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -90,8 +95,10 @@ ::map_to_nodes_sequentially(std::string_view sequence,
     std::vector<node_index> rev_path = map_sequence_to_nodes(*graph_, rev_seq);
 
     // map the forward
-    const auto *dbg_succ = dynamic_cast<const DBGSuccinct*>(graph_.get());
+    const auto *dbg_succ = get_dbg_succ(*graph_);
     if (dbg_succ && get_k() % 2) {
+        const auto *cached = dynamic_cast<const DBGSuccinctCachedView*>(graph_.get());
```

Comment:
So if it's divisible by 2, maybe we should print a warning saying that this thing is gonna be slower?

Reply:
I added this warning to `load_graph.hpp`.